src/util.cpp

```diff
@@ -682,7 +682,36 @@ boost::filesystem::path GetTempPath() {
 
 void runCommand(std::string strCommand)
 {
-    int nErr = ::system(strCommand.c_str());
+    int nErr = 0;
+#ifdef WIN32
+    // If we're in Windows, try using CreateProcess first.
+    ULONG nExitCode;
+    STARTUPINFO info={sizeof(info)};
+    ZeroMemory(&info, sizeof(STARTUPINFO));
+    info.cb = sizeof(STARTUPINFO);
+    info.dwFlags = STARTF_USESHOWWINDOW;
+    info.wShowWindow = SW_HIDE;
+    PROCESS_INFORMATION processInfo;
+    ZeroMemory(&processInfo, sizeof(PROCESS_INFORMATION));
+    if (CreateProcess(NULL, &strCommand[0], NULL, NULL, false, 0, NULL, NULL, &info, &processInfo))
+    {
+        // CreateProcess succeeded so wait, check exit code and clean up.
+        WaitForSingleObject(processInfo.hProcess, INFINITE);
+        if(!GetExitCodeProcess(processInfo.hProcess, &nExitCode))
+          nExitCode = 1;
+        CloseHandle(processInfo.hThread);
+        CloseHandle(processInfo.hProcess);
+        nErr = (int)nExitCode; // Had to pass a ULONG to GetExitCodeProcess().
+    }
+    else
+    {
```

Comment:
Are you sure that this fallback is needed? I'm not sure what the implementation of mingw's system() does internally, but I'd assume it also makes use of CreateProcess or another win32 call.